### ISSUE TITLE
illegal memory access when using pretraining_dataset: , works when swapped to datasets

### ISSUE THREAD

**Issue by lzumot** [axolotl-ai-cloud/axolotl#2418]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

using pretraining dataset in config for pretrain should run normally.

export CUDA_LAUNCH_BLOCKING=1 
export TORCH_USE_CUDA_DSA=1
axolotl train training_scripts/fft.yaml 

### Current behaviour

using pretraining dataset in config for pretrain shows illegal memory access
swapping that to datasets for pretrain works fine

removing liger kernel, CUDA_LAUNCH_BLOCKING=1, reducing context length, increasing grad accum all do not change the error.

datasets version
[ds.txt](https://github.com/user-attachments/files/19286056/ds.txt)

pretrain version
[pt.txt](https://github.com/user-attachments/files/19286057/pt.txt)

### Steps to reproduce

pretraining on 4*H100 ,CUDA 12.7, Driver 565.
notes on swapping datasets and pretraining_dataset in # comments in config

export CUDA_LAUNCH_BLOCKING=1 
export TORCH_USE_CUDA_DSA=1 
axolotl train training_scripts/fft.yaml 

### Config yaml

```yaml
base_model: meta-llama/Llama-3.1-8B

load_in_8bit: false
load_in_4bit: false
strict: false

bf16: auto
fp16:
tf32: false

special_tokens: 
  pad_token: <|finetune_right_pad_id|>

pretraining_dataset: # switch to datasets and it works
  - path: /home/myuser/data/
    data_files: ["/home/myuser/data/train.json",]
    ds_type: json
    split: train 
    text_column: "text"
    type: completion
    trust_remote_code:
    skip: 

eval_datasets:
  - path: /home/myuser/data/
    data_files: ["/home/myuser/data/valid.json",]
    ds_type: json
    split: train 
    text_column: "text"
    type: completion 
    trust_remote_code:
    skip: 

dataset_prepared_path: last_run_prepared 
output_dir: ./outputs/out 

sequence_len: 8192
sample_packing: true 
pad_to_sequence_len: true 
#eval_sample_packing: false 
train_on_inputs: true 
group_by_length: false # set to true when using datasets 

num_epochs: 1 
max_steps: 1

gradient_accumulation_steps: 8 
micro_batch_size: 1

optimizer: adamw_torch_fused 
learning_rate: 2e-5 
lr_scheduler: cosine 
weight_decay: 0.0
warmup_steps: 2


logging_steps: 1
evals_per_epoch: 1 
saves_per_epoch: 1 
flash_attention: true

fsdp:
  - full_shard
  - auto_wrap
fsdp_config:
  fsdp_limit_all_gathers: true
  fsdp_sync_module_states: true
  fsdp_offload_params: true
  fsdp_use_orig_params: false
  fsdp_cpu_ram_efficient_loading: true
  fsdp_auto_wrap_policy: TRANSFORMER_BASED_WRAP
  fsdp_transformer_layer_cls_to_wrap: LlamaDecoderLayer
  fsdp_state_dict_type: FULL_STATE_DICT
  fsdp_sharding_strategy: FULL_SHARD
  fsdp_backward_prefetch: BACKWARD_PRE
  activation_checkpointing: true  

#gradient_checkpointing: true # When using FSDP full shard, instead of using `gradient_checkpointing` in TrainingArguments, please use `activation_checkpointing` in `fsdp_config`. The former introduces a redundant AllGather operation in backward pass. Reference: https://github.com/huggingface/transformers/issues/30404
gradient_checkpointing_kwargs:
  use_reentrant: false

plugins:
  - axolotl.integrations.liger.LigerPlugin
liger_rope: true
liger_rms_norm: true
liger_glu_activation: true
liger_fused_linear_cross_entropy: true
liger_layer_norm: true
```

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.11

### axolotl branch-commit

main/4f5eb42

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
Thanks for making the issue. This was reported on discord here for more info: https://discord.com/channels/1104757954588196865/1111279858136383509/1351156863973986375

Env: bare metal in-house server

**Comment by lzumot** [comment]
the preprocessing seems to start just fine if I use datasets with type: completion instead of pretraining_dataset, with latter it MIGHT sometimes run for a few steps before the error
